fix: ignore errors from iframe content extraction

AryanGupta001 opened the pull request:
using try-except block to handle iframe content extraction error
with catch block Handle the error as needed and continue processing other frames

Closes #2708

barjin commented:
Thank you! Can you port this to [`playwright-utils.ts`](https://github.com/apify/crawlee/blob/c440f5ee0dacaef40cbec19a82372b673494632e/packages/playwright-crawler/src/internals/utils/playwright-utils.ts) as well?

Btw can we wrap everything in the `map` function in `try` (i.e. from line 203 onward)? I'm not sure whether the `frame.contentFrame()` call can throw, but better safe than sorry :)

AryanGupta001 commented:
> Thank you! Can you port this to [`playwright-utils.ts`](https://github.com/apify/crawlee/blob/c440f5ee0dacaef40cbec19a82372b673494632e/packages/playwright-crawler/src/internals/utils/playwright-utils.ts) as well?
> 
> Btw can we wrap everything in the `map` function in `try` (i.e. from line 203 onward)? I'm not sure whether the `frame.contentFrame()` call can throw, but better safe than sorry :)

Sure, I will and update

barjin commented on packages/playwright-crawler/src/internals/utils/playwright-utils.ts:
```diff
@@ -609,18 +609,22 @@ export async function parseWithCheerio(
+                        }, contents);
+                    }
+                } catch(error) {
+                    console.warn('Failed to extract iframe content:', error);
```
Use `log.warning` instead of `console.warn`.

AryanGupta001 replied:
I have made the changes

AryanGupta001 commented:
> Thank you! Can you port this to [`playwright-utils.ts`](https://github.com/apify/crawlee/blob/c440f5ee0dacaef40cbec19a82372b673494632e/packages/playwright-crawler/src/internals/utils/playwright-utils.ts) as well?
> 
> Btw can we wrap everything in the `map` function in `try` (i.e. from line 203 onward)? I'm not sure whether the `frame.contentFrame()` call can throw, but better safe than sorry :)

I have updated the pull request 

Wrapped everything in map function in try block
Added try block in [`playwright-utils.ts`](https://github.com/apify/crawlee/blob/c440f5ee0dacaef40cbec19a82372b673494632e/packages/playwright-crawler/src/internals/utils/playwright-utils.ts) as well

barjin commented:
The last thing - the `biome` formatter is failing now, run `npm run format` in the repo root and commit the changes it makes, that should fix it 👍🏽

barjin reviewed (approved):
Thanks, lgtm! 🚀
